### ISSUE TITLE
Pathfinding issues

### ISSUE THREAD

**Issue by jensj12** [OpenRCT2/OpenRCT2#4668]
**OS:** Windows 10
**Version:** 0.0.5
**Commit/Build:** 2aac1dd

There are some pathfinding issues that the old pathfinding engine would have resolved (by just heading straight towards the goal). Because it looks rather simple, I have no idea how to solve it.
![the openrct group park 4 2016-10-21 12-35-04](https://cloud.githubusercontent.com/assets/13868449/19596112/576ff072-978d-11e6-9371-69b621c0dd8b.png)
The peeps heading for the toilet and hotdog stall are stuck in the red circle. After some time they got bored and tried to leave, but got stuck in the yellow circle heading for the park exit pointed to by the yellow arrow (on the other side of the park). It seems making some paths sloped makes it worse (probably because it is no longer a wide path).
When I got a group of peeps released, they took quite a strange route to the exit, as indicated by the red line.
![the openrct group park 4 2016-10-21 12-35-46](https://cloud.githubusercontent.com/assets/13868449/19596188/a3e690fa-978d-11e6-9c4c-2e79d6e0fe9d.png)

**Comment by duncanspumpkin** [comment]
The strange route on the way out will be because of the wide paths being thinned like that.

**Comment by zaxcav** [comment]
This is correct.  In that path arrangement there is no straight-through route of paths without the wide flag set.
@duncanspumpkin has a far better method of visualising the wide path flags than I do - I hack the code to display path heights on path elements with the wide flag set - I haven't figured out other ways to do it yet.

Here is what the wide paths for one of the pictured wide path intersections looks like (map rotated to provide  clearer view):
![wide1](https://cloud.githubusercontent.com/assets/20344978/19609754/4498b6b8-97d9-11e6-8038-e57b786d9d47.png)

Peeps that have a goal only follow paths without the wide path flag set.
A different thinning process could fix this, e.g. by thinning wide paths to the center so the outside path elements have the wide path flag set, but this is how paths are thinned in vanilla RCT2.

You can force a non-wide path through the intersection by adding an obstacle into it such as follows:
![wide2](https://cloud.githubusercontent.com/assets/20344978/19609761/4cb3c95a-97d9-11e6-9d6c-044072e24f46.png)

This issue doesn't occur for paths less than 3 tiles wide.
So if you make minor paths 1-2 tiles wide and major paths >= 3 tiles wide, then only intersections between major paths need obstacles obstacles in them (reducing the paths through the intersection to less than 3 tiles wide) to provide a straight-through route for the path finding.

**Comment by zaxcav** [comment]
I missed the other issues that are pointed out in the park regarding the peeps stuck on the bridges.
Those heading to the toilet are stuck because the toilet is disconnected from the path.

**Comment by zaxcav** [comment]
I can get the peeps heading for the exit unstuck. I'll submit a PR to fix the peeps heading to the exit soon.

The point to note is that the pathfinding at its core uses the same, original, bad approach to pathfinding from RCT2. I've been doing lots of fixes that result in peeps more successfully getting to their goal, but in doing so I've also been forcing peeps to more stringently follow the result of the path thinning, which itself doesn't produce the most desirable results (as is pointed out in this very issue). I don't intend to also fix the path thinning.  A new implementation of the path finding is planned and shouldn't need path thinning.

@jensj12 I also noticed that the path at 32,84 has gotten disconnected from 32,83 which is restricting access to further up the mountain.
